api/LogicalPathApi.cpp

```diff
@@ -0,0 +1,90 @@
+/**
+*
+* iRODS REST API
+* This is the iRODS REST API
+*
+* OpenAPI spec version: 1.0.0
+* Contact: info@irods.org
+*
+* NOTE: This class is auto generated by the swagger code generator program.
+* https://github.com/swagger-api/swagger-codegen.git
+* Do not edit the class manually.
+*/
+
+#include "LogicalPathApi.h"
+
+#include "constants.hpp"
+
+#include <spdlog/spdlog.h>
+
+namespace io {
+namespace swagger {
+namespace server {
+namespace api {
+
+using namespace io::swagger::server::model;
+
+LogicalPathApi::LogicalPathApi(Pistache::Address addr)
+    : httpEndpoint(std::make_shared<Pistache::Http::Endpoint>(addr))
+{}
+
+void LogicalPathApi::init(size_t thr = 2) {
+    auto opts = Pistache::Http::Endpoint::options()
+        .threads(thr);
+    httpEndpoint->init(opts);
+    setupRoutes();
+}
+
+void LogicalPathApi::start() {
+    httpEndpoint->setHandler(router.handler());
+    httpEndpoint->serve();
+}
+
+void LogicalPathApi::shutdown() {
+    httpEndpoint->shutdown();
+}
+
+void LogicalPathApi::setupRoutes() {
+    using namespace Pistache::Rest;
+
+    Routes::Delete(router, irods::rest::base_url + "/logicalpath", Routes::bind(&LogicalPathApi::handler, this));
```

Comment:
We should consider a PUT or POST request to this `/logicalpath` endpoint to be an 'alias' for `/stream` PUT?

Hmm....

Reply:
Just a note: I'm pushing to this branch just because my workflow often involves pulling from github within the container. This is under active editing.

Reply:
In that case, would you mind converting this PR to a draft? That way it's markedly in-progress and also removes the possibility of accidentally merging it on my part :)